data-extraction/pyDataExtraction/commonTypes/Graph.py

```diff
@@ -0,0 +1,76 @@
+from typing import Union, Dict, Optional
+from pyDataExtraction.commonTypes.base import DataType
+
+# from docs
+"""
+interface NodeGraphData {
+	id: string;
+	label?: string;
+	color?: string;
+	shape?: "ellipse" | "box";
+}
+"""
+"""
+interface EdgeGraphData {
+	from: string;
+	to: string;
+	label?: string;
+	id?: string;
+	color?: string;
+	dashes?: boolean;
+}
+"""
+# NOTE: may not be able to encapsulate edge in separate class due to from being a syntax token
+"""
+class Edge:
+    
+    def __init__(self,from: str, to: str,):
+        self.fromnode
+        
+    def __repr__(self):
+        pass
+"""
+# NOTE: ran into issue Node object is not json serializable when ecapsulating in own class
```

Comment:
We can add a method in Node class to serialize the data into json.

Reply:
thanks, I'm just now starting to actually learn about serialization due to this project and a personal rust project.

could you point me in the direction of some documentation on how to do so?

Reply:
@skewballfox you can create a dictionary of the desired form and serialize it into JSON:
```
def to_json(self):
    return json.dumps({'from': self.from_node, 'to': self.to})
```

Reply:
`to_node` would also be more readable.

Reply:
@insolor this is similar to what I was doing in `__repr__` but that definitely helps fix that issue in a way that doesn't involve mangling the language. I don't know why overloading `from` was my first guess.

also I think I figured out why `Node` wasn't serializable. It was because the `__repr__` function doesn't seem to be recursive. as an example When calling `print(graph)` that doesn't automatically behave as if `print(node)` were called.

either I could come up with a generic method that does this in `DataType` to avoid having to define it in every current and future dataType that uses other objects as components(if present), or we have to overload the function in those cases. 

I think I might have an idea using `__store__`, I'll have to see though